src/types/insights.ts

```diff
@@ -0,0 +1,16 @@
+export type InsightsClientMethod =
+  | 'clickedObjectIDsAfterSearch'
+  | 'convertedObjectIDsAfterSearch';
+
+export type InsightsClientPayload = {
+  eventName?: string;
+  queryID?: string;
+  index?: string;
+  objectIDs?: string[];
+  positions?: number[];
+};
+
+export type InsightsClient = (
```

Comment:
We could enforce which `payload` match which `eventName` with an overload. That way we can have more specific `payload` with less optionals.

Reply:
As discussed offline I ended giving up on overloading for simplicity.